src/Components/WebAssembly/Build/src/targets/Compression.targets

```diff
@@ -0,0 +1,61 @@
+<Project>
+  <PropertyGroup>
+
+    <ResolveCurrentProjectStaticWebAssetsDependsOn>
+      $(ResolveCurrentProjectStaticWebAssetsDependsOn);
+      _CompressBlazorApplicationFiles;
+    </ResolveCurrentProjectStaticWebAssetsDependsOn>
+
+  </PropertyGroup>
+
+  <Target Name="_ResolveAssetsToCompress" AfterTargets="_ResolveBlazorGeneratedAssets">
+
+    <PropertyGroup>
+      <StaticResourcesIntermediateOutputPath>$(IntermediateOutputPath)compressed\</StaticResourcesIntermediateOutputPath>
+    </PropertyGroup>
+
+    <MakeDir Directories="$(StaticResourcesIntermediateOutputPath)" />
+    <ItemGroup>
+      <_StaticResourceToCompress Include="@(StaticWebAsset)" Condition="'%(SourceType)' == '' and $([System.String]::Copy('%(RelativePath)').Replace('\','/').StartsWith('_framework/'))" KeepDuplicates="false">
+        <TargetCompressionPath>$(StaticResourcesIntermediateOutputPath)%(RelativePath).gz</TargetCompressionPath>
+        <TargetOutputPath>%(RelativePath).gz</TargetOutputPath>
+        <RelativePath>%(RelativePath).gz</RelativePath>
+        <InputSource>%(FullPath)</InputSource>
+      </_StaticResourceToCompress>
+
+      <_StaticResourceToCompress Condition="'$(BlazorLinkOnBuild)' == 'true' and ('%(Extension)' == '.dll' or '%(Extension)' == '.pdb')">
```

Comment:
Could we include `.wasm` and `.js` too?

Reply:
TBH compressing `pdb` is probably unnecessary, as it's purely a dev-time concept.

Reply:
They are included, this is just creating a different input for the compression task to support incrementalism with linking

Reply:
This itemgroup is just updating the input source property for pdbs and dlls to point to a hash file based on the output contents.

The main reason being that the linker will always touch the files, which breaks any potential incrementalism when linking.

Reply:
I'll add comments to explain this for future reference.

Reply:
Ah I see. So for now we're compressing everything in `_framework/**`, which is OK because we don't have any pre-compressed file types (e.g., jpg, etc.). Right?

Reply:
Yes. Once we figure out the static files issue we were looking at yesterday, we can do this for the entire app.

Reply:
Exactly.